geospatial/dted.ksy

```diff
@@ -0,0 +1,596 @@
+        doc: |
+          Recognition Sentinel.
+      - id: data_block_count
+        type: b24
+        doc: |
+          Sequential count of the block within the file, starting with zero |
+          for the first block (Fixed Binary). (Data block count)
+      - id: logitude_count
+        type: u2
+        doc: |
+          Count of the meridian. True longitude = longitude count x data |
+          interval + origin (Offset from the SW corner longitude) |
+          (Fixed Binary).
+      - id: latitude_count
+        type: u2
+        doc: |
+          Count of the parallel. True latitude = latitude count x data |
+          interval + origin (Offset from the SW corner latitude) |
+          (Fixed Binary).
+      - id: elevations
+        type: data_record_elevation
+        repeat: expr
+        repeat-expr: _parent.uhl.number_of_latitude_points.value
+        doc: |
+          True elevation value of point N of meridian in meters (Fixed Binary).
+      - id: checksum
+        type: s4
+        doc: |
+          Algebraic addition of contents of block. Sum is computed as an 
+          integer summation of 8-bit values (Fixed Binary).
+  data_record_elevation:
+    doc: |
+      type which repesents a single elevation value with it's unique encoding
+      mechanism.
+    seq:
+      - id: is_negative
+        type: b1
+      - id: value
+        type: b15
+    instances:
+      meters:
+        value: value * (is_negative? -1 : 1)
+  ascii_int:
+    seq:
+      - id: text
+        type: str
+        size: 4
+    instances:
+      value:
+        value: text.to_i
+  coordinate_pair_ddmmss_sh:
+    doc: |
+      Typed used to make types consistent once parsed
+    seq:
+      - id: latitude
+        type: angle_ddmmss_sh
+      - id: longitude
+        type: angle_dddmmss_sh
+  coordinate_pair_ddmmssh:
+    doc: |
+      Typed used to make types consistent once parsed
+    seq:
+      - id: latitude
+        type: angle_ddmmssh
+      - id: longitude
+        type: angle_dddmmssh
+  angle_ddmmss_sh:
+    doc: |
+      Typed used created to convert to degree decimal representation
+    seq:
+      - id: degrees
+        type: str
+        size: 2
+      - id: minutes
+        type: str
+        size: 2
+      - id: seconds
+        type: str
+        size: 4
+      - id: hemisphere
+        type: str
+        size: 1
+    instances:
+      degree_decimal:
+        value: (degrees.to_i + minutes.to_i/60.0 + seconds.to_i/3600.0) * (hemisphere == 'N'? 1 : -1)
+  angle_dddmmss_sh:
+    doc: |
+      Typed used created to convert to degree decimal representation
+    seq:
+      - id: degrees
+        type: str
+        size: 3
+      - id: minutes
+        type: str
+        size: 2
+      - id: seconds
+        type: str
+        size: 4
+      - id: hemisphere
+        type: str
+        size: 1
+    instances:
+      degree_decimal:
+        value: (degrees.to_i + minutes.to_i/60.0 + seconds.to_i/3600.0) * (hemisphere == 'N'? 1 : -1)
```

Comment:
On this field the `seconds.to_i` is technically incorrect.  The underlying text is actually a floating -point number.   However I didn't see any methods in the docs to be able to parse the text as such.  I was expecting something like `seconds.to_f`.  Unless there is a better way I'll change it as follows:
```
  angle_dddmmss_sh:
    doc: |
      Typed used created to convert to degree decimal representation
    seq:
      - id: degrees
        type: str
        size: 3
      - id: minutes
        type: str
        size: 2
      - id: seconds
        type: str
        size: 2
      - id: decimal_point
        type: str
        size: 1
      - id: tenths_of_seconds
        type: str
        size: 1
      - id: hemisphere
        type: str
        size: 1
    instances:
      value:
        value: (degrees.to_i + minutes.to_i/60.0 + seconds.to_i/3600.0 + tenths_of_seconds.to_i/36000.0) * (hemisphere == 'N'? 1 : -1)
        doc: |
          Degree decimal floating point representation     
```

Reply:
https://github.com/kaitai-io/kaitai_struct/issues/524

Reply:
Cool until then I'll leave it as shown above.